flink-core/src/test/java/org/apache/flink/configuration/DelegatingConfigurationTest.java

```diff
@@ -88,4 +90,49 @@ private String typeParamToString(Class<?>[] classes) {
 			assertTrue("Foo method '" + configurationMethod.getName() + "' has not been wrapped correctly in DelegatingConfiguration wrapper", hasMethod);
 		}
 	}
+	
+	@Test
+	public void testDelegationConfigurationWithNullPrefix() {
+
+		Configuration backingConf = new Configuration();
+		backingConf.setValueInternal("test-key", "value");
+
+		DelegatingConfiguration configuration = new DelegatingConfiguration(
+				backingConf, null);
+		Set<String> keySet = configuration.keySet();
+
+		assertEquals(keySet, backingConf.keySet());
+
+	}
+
+	@Test
+	public void testDelegationConfigurationWithPrefix() {
+
+		String prefix = "pref-";
+
+		/*
+		 * Key matches the prefix
+		 */
+		Configuration backingConf = new Configuration();
+		backingConf.setValueInternal("pref-key", "value");
```

Comment:
I think `"pref-"` can be replaced by the `prefix` variable, just to make it consistent.